fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/ExistsValue.java

```diff
@@ -54,32 +54,25 @@
  * A {@link Value} that checks whether an item exists in its child quantifier expression or not.
  */
 @API(API.Status.EXPERIMENTAL)
-public class ExistsValue extends AbstractValue implements BooleanValue, ValueWithChild, Value.CompileTimeValue {
+public class ExistsValue extends AbstractValueWithChild implements BooleanValue, Value.CompileTimeValue {
     private static final ObjectPlanHash BASE_HASH = new ObjectPlanHash("Exists-Value");
-    @Nonnull
-    private final QuantifiedObjectValue child;
 
     public ExistsValue(@Nonnull QuantifiedObjectValue child) {
-        this.child = child;
+        super(child);
     }
 
     @Override
     @SuppressWarnings({"java:S2637", "ConstantConditions"}) // TODO the alternative component should not be null
     @SpotBugsSuppressWarnings("NP_NONNULL_PARAM_VIOLATION")
     public Optional<QueryPredicate> toQueryPredicate(@Nullable final TypeRepository typeRepository,
                                                      @Nonnull final CorrelationIdentifier innermostAlias) {
-        return Optional.of(new ExistsPredicate(child.getAlias()));
+        return Optional.of(new ExistsPredicate(((QuantifiedObjectValue)getChild()).getAlias()));
```

Comment:
AbstractValueWithChild is defining a child but there is no proto message reflecting that. Thus every extending class needs to deal with the child in their own `toProto` and `fromProto` which is something we absolutely want to avoid as a 1-1 relationship between proto and pojo is necessary to not go insane.
So you can either define a proto message `PAbstractValueWithChild` and use the usual pattern with a  `super`  field for all extending classes, or you keep these super classes interfaces for now.

Reply:
reverted interface-to-abstract-class changes.